app/lib/actions/candlepin/candlepin_listening_service.rb

```diff
@@ -40,7 +40,9 @@ def close
       end
 
       def retrieve
-        return @receiver.fetch(TIMEOUT)
+        result = @receiver.fetch(TIMEOUT)
+        safe_release(result) if result
```

Comment:
My knowledge is limited, but isn't `safe_release` a good candidate for ensure block? In case there is an exception in any client code (or future code added here) the message would be acknowledged.

Reply:
Seems reasonable for future proofing 👍